src/core/mapparser.cpp

```diff
@@ -0,0 +1,116 @@
+                    (static_cast<unsigned char>(in.at(7)) << 24);
+    int mapPrimaryTilesetNum = static_cast<unsigned char>(in.at(8)) |
+                               (static_cast<unsigned char>(in.at(9)) << 8) |
+                               (static_cast<unsigned char>(in.at(10)) << 16) |
+                               (static_cast<unsigned char>(in.at(11)) << 24);
+    int mapSecondaryTilesetNum = static_cast<unsigned char>(in.at(12)) |
+                                 (static_cast<unsigned char>(in.at(13)) << 8) |
+                                 (static_cast<unsigned char>(in.at(14)) << 16) |
+                                 (static_cast<unsigned char>(in.at(15)) << 24);
+
+    /*int maxMetatiles = primaryTileset ? Project::getNumMetatilesPrimary() : Project::getNumMetatilesTotal() - Project::getNumMetatilesPrimary();
+    int numMetatiles = static_cast<unsigned char>(in.at(0)) |
+                                (static_cast<unsigned char>(in.at(1)) << 8) |
+                                (static_cast<unsigned char>(in.at(2)) << 16) |
+                                (static_cast<unsigned char>(in.at(3)) << 24);
+    if (numMetatiles > maxMetatiles) {
+        *error = true;
+        logError(QString(".bvd file contains data for %1 metatiles, but the maximum number of metatiles is %2.").arg(numMetatiles).arg(maxMetatiles));
+        return nullptr;
+    }
+
+    if (numMetatiles < 1) {
+        *error = true;
+        logError(QString(".bvd file contains no data for metatiles."));
+        return nullptr;
+    }*/
+
+    int numMetatiles = mapWidth * mapHeight;
+    int expectedFileSize = 20 + (numBorderTiles * 2) + (numMetatiles * 2);
+    if (in.length() != expectedFileSize) {
+        *error = true;
+        logError(QString(".map file is an unexpected size. Expected %1 bytes, but it has %2 bytes.").arg(expectedFileSize).arg(in.length()));
+        return nullptr;
+    }
+
+    Blockdata *blockdata = new Blockdata();
+    for (int i = mapDataOffset; (i + 1) < in.length(); i += 2) {
+        uint16_t word = static_cast<uint16_t>((in[i] & 0xff) + ((in[i + 1] & 0xff) << 8));
+        blockdata->addBlock(word);
+    }
+
+    Blockdata *border = nullptr;
+    if (numBorderTiles != 0) {
+        border = new Blockdata();
+        for (int i = 20; (i + 1) < 20 + (numBorderTiles * 2); i += 2) {
+            uint16_t word = static_cast<uint16_t>((in[i] & 0xff) + ((in[i + 1] & 0xff) << 8));
+            border->addBlock(word);
+        }
+    }
+
+    MapLayout *mapLayout = new MapLayout();
+    mapLayout->width = QString::number(mapWidth);
+    mapLayout->height = QString::number(mapHeight);
+    mapLayout->border_width = (borderWidth == 0) ?  QString::number(2) :  QString::number(borderWidth);
+    mapLayout->border_height = (borderHeight == 0) ?  QString::number(2) :  QString::number(borderHeight);
+
+    QList<QString> tilesets = project->tilesetLabelsOrdered;
+
+    if (mapPrimaryTilesetNum > tilesets.size())
```

Comment:
Rather than assuming the tilesets are ordered the same between the two projects, would it make more sense to ask the user for the tileset, instead?

Reply:
I'm not saying we should do it one way or the other.  I'm just asking the question.